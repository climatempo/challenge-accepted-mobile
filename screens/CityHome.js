import React, { useState, useEffect } from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { Title, Card } from 'react-native-paper';
import axios from 'axios'
import CardDia from './CardDia'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { TOKENAPI } from '../utils'

const CityHome = (props) => {

    const [cidade, setCidade] = useState({})
    const [loading, setLoading] = useState(true)
    const [error, setError] = useState({ hasError: false, message: '' })

    useEffect(() => {
        getDataFetch()
    }, [props.route.params])



    const getDataFetch = async () => {

        const { idLocale } = props.route.params

        try {
            const { data } = await axios.get(`http://apiadvisor.climatempo.com.br/api/v1/forecast/locale/${idLocale}/days/15?token=${TOKENAPI}`)
            await AsyncStorage.setItem('idCidade', JSON.stringify(idLocale))
            setCidade(data)
            setLoading(false)
        } catch (error) {
            const { data } = await axios.get(`http://apiadvisor.climatempo.com.br/api/v1/locale/city/${idLocale}?token=${TOKENAPI}`)
            setError({ hasError: true, message: `Sem permissão para localizar as informações da cidade de ${data.name}/${data.state} ! Tente Lins-SP` })
            setLoading(false)
            console.log('Deu erro: ' + error)
        }
    }


    return (
        <>
            {error.hasError ?
                <View style={styles.container}>
                    <Title style={styles.errormsg}>{error.message}</Title>
                </View>
                :
                !loading ?

                    <View style={styles.container}>
                        <Card>
                            <Title style={styles.title}>{cidade.name}/{cidade.state}</Title>
                        </Card>
                        <ScrollView>
                            {cidade.data.map(dia => (
                                <CardDia dia={dia} key={dia.date} />
                            ))}
                        </ScrollView>
                    </View>

                    :

                    <Title>Carregando...</Title>
            }
        </>
    )


}

const styles = StyleSheet.create({
    container: {
        flex: 1
    },
    title: {
        textAlign: 'center'
    },
    errormsg: {
        marginTop: 5,
        textAlign: 'justify'
    }
});

export default CityHome