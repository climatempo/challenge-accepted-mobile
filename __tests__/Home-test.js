import 'react-native';
import React from 'react';
import CityHome from '../screens/CityHome';
import renderer from 'react-test-renderer'

test('Should render CityHome component', () => {

    const idLocale = 3906

    const snap = renderer.create(
        <CityHome route={idLocale} />
    ).toJSON();

    expect(snap).toMatchSnapshot();
})