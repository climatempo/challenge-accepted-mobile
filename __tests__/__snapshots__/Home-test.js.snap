// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CityHome component 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "System",
          "fontWeight": "500",
          "textAlign": "left",
          "writingDirection": "ltr",
        },
        Array [
          Object {
            "fontSize": 20,
            "letterSpacing": 0.15,
            "lineHeight": 30,
            "marginVertical": 2,
          },
          undefined,
        ],
      ],
    ]
  }
>
  Carregando...
</Text>
`;
